File: problems/CSES/Sorting_and_Searching/tasks_and_deadlines/tasks_and_deadlines.py
import sys
# sys.setrecursionlimit(200000)

input = sys.stdin.readline

def main():
    n = int(input())
    tasks = [list(map(int, input().split())) for _ in range(n)]

    tasks.sort(key=lambda x: x[0])

    time = 0
    res = 0

    for a, d in tasks:
        time += a
        res += d - time

    print(res)

main()

File: problems/CSES/Sorting_and_Searching/tasks_and_deadlines/tasks_and_deadlines.cpp
#include <bits/stdc++.h>
using namespace std;
using ll = long long;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;

    vector<pair<ll, ll>> tasks(n);
    for (int i = 0; i < n; i++)
        cin >> tasks[i].first >> tasks[i].second;
        
    sort(tasks.begin(), tasks.end());
    
    ll time = 0;
    ll res = 0;

    for (const auto& [a, d] : tasks) {
        time += a;
        res += d - time;
    }

    cout << res << "\n";

    return 0;
}
